appdev/guides/dependencies.rst

```diff
@@ -92,45 +88,66 @@ If the library is not in the Ubuntu Repositories, but in a PPA, you can add the
+command, which provides you with a target to link against or variables pointing
+to the library's binary and include directories. The additional lines on your
+CMakeLists.txt then may look like:
 
 .. code-block:: CMake
 
-    execute_process(
-        COMMAND dpkg-architecture -qDEB_HOST_MULTIARCH
-        OUTPUT_VARIABLE ARCH_TRIPLET
-        OUTPUT_STRIP_TRAILING_WHITESPACE
-    )
-    set(SOMELIBRARY_DIR "${CMAKE_SOURCE_DIR}/build/somelib/${ARCH_TRIPLET}")
     find_package(SOMELIBRARY REQUIRED)
+
     include_directories(${SOMELIBRARY_INCLUDE_DIRS})
     target_link_libraries(mytarget ${SOMELIBRARY_LIBS})
 
-The command ``dpkg-architecture -qDEB_HOST_MULTIARCH`` is used to query the target architecture, which is part of the library build directory path, if you compiled the library with Clickable.
-
-We define the variable ``SOMELIBRARY_DIR`` with the path to the libraries build directory, to help CMake find the configuration of the library named ``SOMELIBRARY``. You may not need to do this, if you installed the library from the Ubuntu Repositories.
+The ``find_package`` command usually defines the path to the include directory
+as ``SOMELIBRARY_INCLUDE_DIRS`` and the library's binaries as
+``SOMELIBRARY_LIBS`` (check on the library's documentation on what their CMake
+configuration provides exactly). Use them with the ``include_directories`` and
+``target_link_libraries`` commands. See the
+`Camera Scanner ImageProcessing CMakeLists.txt <https://github.com/jonnius/camera-scanner/blob/master/plugins/ImageProcessing/CMakeLists.txt#L23>`_ for a real world example.
 
-The ``find_package`` command defines the path to the include directory as ``SOMELIBRARY_INCLUDE_DIRS`` and the library's binaries as ``SOMELIBRARY_LIBS``. We use those with the ``include_directories`` and ``target_link_libraries`` commands. See the `Camera Scanner ImageProcessing CMakeLists.txt <https://github.com/jonnius/camera-scanner/blob/master/plugins/ImageProcessing/CMakeLists.txt#L23>`_ for a real world example.
+At this point you should already be able to test with Clickable's desktop mode
+running ``clickable desktop``.
 
 Deployment
 ----------
-If you link a library statically with your app, you do not need to ship the library explicitly, as it is already inside your app binary. To do so, you usually need to compile the library yourself. Otherwise, continue with this section.
+If you linked the library statically, you can skip this step, as the library's
+binary is already inside your app's binary.
 
-Find out which components you need to ship. Usually this is one or more ``*.so`` (shared objects) files, linked dynamically. To get the files into the click package, you need to add an ``install`` command to your build configuration. Add the following lines to your CMakeLists.txt:
+Find out which components you need to ship (e.g. by `printing <https://cmake.org/cmake/help/v3.0/command/message.html>`_ the CMake variable ``${SOMELIBRARY_LIBS}``). Usually
```

Comment:
Maybe an example of printing the variable would be useful? I'm not sure what you're trying to tell the reader to do.

Reply:
I replaced the hint by another method to find the missing pieces _[no comma]_ that should be easier to follow.